### DISCUSSION TITLE
create a groupBy based on month on dateField query using prisma

### DISCUSSION THREAD

**Discussion by RuchitPatel14** [prisma/prisma#22088] [Q&A]
### Question

I want to create a groupBy query using Prisma
```sql
select id, count(id), DATE_FORMAT(due_start_date, '%d %b %Y') from activity group by DAY(due_start_date), MONTH(due_start_date), YEAR(due_start_date)
```
so inside the prisma.groupBy we can only pass by: ['due_start_date'] so is there any method or something?

- prisma: '5.4.1'

**Comment by ludralph** [comment]
Hi @RuchitPatel14 👋 

Thank you for raising this question.

Prisma does not support grouping by date parts (like day, month, year) in the `groupBy` method. The suggestion for adding datetime operators to the `groupBy` key, such as grouping by year or by month, is discussed in this open GitHub issue. 

- https://github.com/prisma/prisma/issues/6653

Additionally, SQL functions like `DATE_FORMAT` do not have a representation in Prisma. In this case, you will need to use the [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) method to execute raw SQL queries that include the `DATE_FORMAT` function and other date-related grouping operations. You can also leverage [SafeQL](https://www.prisma.io/docs/guides/other/advanced-database-tasks/typesafe-raw-sql/safeql#what-is-safeql) to create type safe raw queries.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Reply by axelmy318** [reply]
I think it's clear that this feature is primordial to a lot of people and still not developed.
